File: WebSite/EulerAList/pset26.cpp
#include <iostream>     // std::cout
#include <algorithm>    // std::next_permutation, std::sort
#include <cmath>
#include <string>
#include <vector>
#include <map>

using namespace std;

string fractionToDecimal(int denr) 
{ 
    string res;
    map <int, int> mp; 
    mp.clear(); 
    int rem = 1; 
    while ( (rem!=0) && (mp.find(rem) == mp.end()) ) 
    { 
        mp[rem] = res.length(); 
        rem = rem*10; 
        int res_part = rem / denr; 
        res += to_string(res_part); 
        rem = rem % denr; 
    }   
    return (rem == 0)? "" : res.substr(mp[rem]); 
} 

int main () 
{
    int result = 0;
    int maxLen = 0;
    for (int i = 2; i < 1000; i ++)
    {
        if (fractionToDecimal(i).size() > maxLen)
        {
            maxLen = fractionToDecimal(i).size();
            result = i;
        }
    }
    cout << result << endl;
}

File: WebSite/EulerAList/pset26.py
def fractionToDecimal(denr):
    res = ""
    mp = dict() 
    rem = 1 
    while (rem!=0) and not rem in mp.keys(): 
        mp[rem] = len(res) 
        rem *= 10
        res_part = rem // denr 
        res = res + str(res_part) 
        rem = rem % denr
    if rem == 0:
        return ""
    else:
        return res[mp[rem]:-1]

result = 0
maxLen = 0
for i in range(2, 1000):
    if len(fractionToDecimal(i)) > maxLen:
        maxLen = len(fractionToDecimal(i))
        result = i
print(result)